Panic during jx create spring

Using:

> NAME               VERSION
> jx                 1.2.20
> jenkins x platform 0.0.795
> kubernetes cluster v1.9.3-gke.0
> kubectl            v1.9.6
> helm client        v2.8.2+ga802316
> helm server        v2.8.2+ga802316
> git                git version 2.15.1 (Apple Git-101)

Actions:
`jx create spring -t 1.5.12.RELEASE -d web -d actuator`

```
About to create repository deleteme20 on server github.com with user dvsbot
? Which organisation do you want to use? dvsbot
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16dba8a]

goroutine 1 [running]:
github.com/jenkins-x/jx/pkg/gits.(*GitHubProvider).ValidateRepositoryName(0xc42058a240, 0xc420447310, 0x6, 0xc42033a04c, 0xa, 0x1f, 0xc42033a04c)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/gits/github.go:536 +0x9a
github.com/jenkins-x/jx/pkg/gits.PickNewGitRepository.func2(0x23c5ce0, 0xc4205534a0, 0xc4205534a0, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/gits/git_repo.go:144 +0x128
github.com/jenkins-x/jx/vendor/gopkg.in/AlecAivazis/survey%2ev1.Ask(0xc42084d648, 0x1, 0x1, 0x2376be0, 0xc420553360, 0x30, 0x251e2c0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/gopkg.in/AlecAivazis/survey.v1/survey.go:105 +0x1d9
github.com/jenkins-x/jx/vendor/gopkg.in/AlecAivazis/survey%2ev1.AskOne(0x32ee680, 0xc42031e400, 0x2376be0, 0xc420553360, 0xc42060f020, 0x0, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/gopkg.in/AlecAivazis/survey.v1/survey.go:54 +0xbe
github.com/jenkins-x/jx/pkg/gits.PickNewGitRepository(0x32df980, 0xc42000c018, 0xc42033a000, 0xc42033cf80, 0x1d, 0xc420281f80, 0xc42033a04c, 0xa, 0xc4200b43e8, 0xc420526120, ...)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/gits/git_repo.go:146 +0xbfc
github.com/jenkins-x/jx/pkg/jx/cmd.(*ImportOptions).CreateNewRemoteRepository(0xc4200b42c0, 0x0, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/import.go:549 +0x172
github.com/jenkins-x/jx/pkg/jx/cmd.(*ImportOptions).Run(0xc4200b42c0, 0x0, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/import.go:328 +0x46f
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateProjectOptions).ImportCreatedProject(0xc4200b42c0, 0xc42033a030, 0x26, 0xc42057dc48, 0x1)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create.go:99 +0x6a
github.com/jenkins-x/jx/pkg/jx/cmd.(*CreateSpringOptions).Run(0xc4200b42c0, 0xc42006bc50, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_spring.go:123 +0x2f2
github.com/jenkins-x/jx/pkg/jx/cmd.NewCmdCreateSpring.func1(0xc420156000, 0xc42014ae40, 0x0, 0x6)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/create_spring.go:71 +0x75
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).execute(0xc420156000, 0xc42014ade0, 0x6, 0x6, 0xc420156000, 0xc42014ade0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200c4f00, 0xc420157180, 0xc42060cf00, 0xc4204a4500)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:852 +0x334
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200c4f00, 0xc42074d6c0, 0x32df900)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/jenkins-x/jx/cmd/jx/app.Run(0xc4204ebf70, 0x228a429)
	/home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/app/jx.go:18 +0x90
main.main()
	/home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/jx.go:10 +0x22
```

--- a/pkg/gits/bitbucket.go
+++ b/pkg/gits/bitbucket.go
@@ -263,7 +263,7 @@ func (b *BitbucketCloudProvider) ValidateRepositoryName(org string, name string)
 		name,
 	)
 
-	if r.StatusCode == 404 {
+	if r != nil && r.StatusCode == 404 {
 		return nil
 	}
 
@@ -357,7 +357,7 @@ func (b *BitbucketCloudProvider) UpdatePullRequestStatus(pr *GitPullRequest) err
 	bitbucketPR, _, err := b.Client.PullrequestsApi.RepositoriesUsernameRepoSlugPullrequestsPullRequestIdGet(
 		b.Context,
 		b.Username,
-		strings.TrimPrefix(pr.Repo, b.Username + "/"),
+		strings.TrimPrefix(pr.Repo, b.Username+"/"),
 		prID,
 	)
 
@@ -379,7 +379,7 @@ func (b *BitbucketCloudProvider) UpdatePullRequestStatus(pr *GitPullRequest) err
 		b.Context,
 		b.Username,
 		strconv.FormatInt(int64(prID), 10),
-		strings.TrimPrefix(pr.Repo, b.Username + "/"),
+		strings.TrimPrefix(pr.Repo, b.Username+"/"),
 	)
 
 	if err != nil {
@@ -412,7 +412,7 @@ func (b *BitbucketCloudProvider) PullRequestLastCommitStatus(pr *GitPullRequest)
 		result, _, err := b.Client.CommitstatusesApi.RepositoriesUsernameRepoSlugCommitNodeStatusesGet(
 			b.Context,
 			b.Username,
-			strings.TrimPrefix(pr.Repo, b.Username + "/"),
+			strings.TrimPrefix(pr.Repo, b.Username+"/"),
 			pr.LastCommitSha,
 		)
 
@@ -489,7 +489,7 @@ func (b *BitbucketCloudProvider) MergePullRequest(pr *GitPullRequest, message st
 		b.Context,
 		b.Username,
 		strconv.FormatInt(int64(*pr.Number), 10),
-		strings.TrimPrefix(pr.Repo, b.Username + "/"),
+		strings.TrimPrefix(pr.Repo, b.Username+"/"),
 		options,
 	)
 

--- a/pkg/gits/github.go
+++ b/pkg/gits/github.go
@@ -533,7 +533,7 @@ func (p *GitHubProvider) ValidateRepositoryName(org string, name string) error {
 	if err == nil {
 		return fmt.Errorf("Repository %s already exists", GitRepoName(org, name))
 	}
-	if r.StatusCode == 404 {
+	if r != nil && r.StatusCode == 404 {
 		return nil
 	}
 	return err
@@ -543,7 +543,7 @@ func (p *GitHubProvider) UpdateRelease(owner string, repo string, tag string, re
 	release := &github.RepositoryRelease{}
 	rel, r, err := p.Client.Repositories.GetReleaseByTag(p.Context, owner, repo, tag)
 
-	if r.StatusCode == 404 && !strings.HasPrefix(tag, "v") {
+	if r != nil && r.StatusCode == 404 && !strings.HasPrefix(tag, "v") {
 		// sometimes we prepend a v for example when using gh-release
 		// so lets make sure we don't create a double release
 		vtag := "v" + tag
@@ -570,7 +570,7 @@ func (p *GitHubProvider) UpdateRelease(owner string, repo string, tag string, re
 	if release.Body == nil && releaseInfo.Body != "" {
 		release.Body = &releaseInfo.Body
 	}
-	if r.StatusCode == 404 {
+	if r != nil && r.StatusCode == 404 {
 		fmt.Printf("No release found for %s/%s and tag %s so creating a new release\n", owner, repo, tag)
 		_, _, err = p.Client.Repositories.CreateRelease(p.Context, owner, repo, release)
 		return err
@@ -589,7 +589,7 @@ func (p *GitHubProvider) UpdateRelease(owner string, repo string, tag string, re
 
 func (p *GitHubProvider) GetIssue(org string, name string, number int) (*GitIssue, error) {
 	i, r, err := p.Client.Issues.Get(p.Context, org, name, number)
-	if r.StatusCode == 404 {
+	if r != nil && r.StatusCode == 404 {
 		return nil, nil
 	}
 	if err != nil {
@@ -621,7 +621,7 @@ func (p *GitHubProvider) searchIssuesWithOptions(org string, name string, opts *
 	answer := []*GitIssue{}
 	for {
 		issues, r, err := p.Client.Issues.ListByRepo(p.Context, org, name, opts)
-		if r.StatusCode == 404 {
+		if r != nil && r.StatusCode == 404 {
 			return answer, nil
 		}
 		if err != nil {

--- a/pkg/gits/gitlab.go
+++ b/pkg/gits/gitlab.go
@@ -163,7 +163,7 @@ func (g *GitlabProvider) ValidateRepositoryName(org, name string) error {
 	if err == nil {
 		return fmt.Errorf("repository %s already exists", pid)
 	}
-	if r.StatusCode == 404 {
+	if r != nil && r.StatusCode == 404 {
 		return nil
 	}
 	return err

